python/grass/temporal/register.py

```diff
@@ -135,23 +135,17 @@ def register_maps_in_space_time_dataset(
             dbif.close()
             msgr.fatal(
                 _(
-                    "Space time %(sp)s dataset <%(name)s> with relative"
-                    " time found, but no relative unit set for %(sp)s "
+                    "Space time {sp} dataset <{name}> with relative"
+                    " time found, but no relative unit set for {sp} "
                     "maps"
-                )
-                % {"name": name, "sp": sp.get_new_map_instance(None).get_type()}
+                ).format(name=name, sp=sp.get_new_map_instance(None).get_type())
             )
 
     maplist = []
 
     # Map names as comma separated string
     if maps:
-        if maps.find(",") < 0:
-            maplist = [
-                maps,
-            ]
-        else:
-            maplist = maps.split(",")
+        maplist = maps.split(",")
```

Comment:
The function doc does not fully specify. It says: `:param maps: A comma separated list of map names`. But the comment right above th "if maps" block says: `# Map names as comma separated string`
So I suppose the maps argument should be a string.

Reply:
One of my ideas that I only explored for two weekends near the Pylint-pytest CI failures, was to start annotating where we could various places of the code. That way IDEs and type checkers (multiple exist, not only mypy that is very hard to satisfy, like the one from Microsoft (pyright), the one from Facebook (pyre), or the one from google (pytype).
But we have to start from the base and build up on it. After that, it would be obvious what is allowed or not. But that's a multi-month gradual effort, and we already have the project-location, and pytest going on (plus my personal goal of code coverage).

Reply:
Thanks, @echoix for the feedback. I agree that type definitions in all functions would be a valuable improvement of the developer documentation. We would need some style guidelines for that, I guess. We can start incremental by adding those where we make changes anyway. However, I would suggest opening a separate issue for that, even if we may not tackle that as priority 1 right away...

Besides that, do you see a need for more changes in this PR? I would like to get it ready to merge as the lack of usability of _creation_time_ starts hampering my TGIS use more and more...

Reply:
I don't think any of what I found out here was really critical. But I don't understand enough on what's going here to know if it's the good thing to do. Or if there's missing something. I made an effort some time ago to really review it since you asked some and no one answered. I could unblock you if you really need though

Reply:
> I could unblock you if you really need though

Thanks, @echoix ! I really appreciate your effort and support! Tests pass and I cannot see reasons for a performance regression, so I think unblocking and merging should be OK...

Again, thanks for looking into this!

Reply:
Ok, let's go with it then